src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/AvailablePlugin.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.Toolkit.Plugins.Core;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Core.Packaging.Metadata;
+using Microsoft.Performance.Toolkit.Plugins.Core.Transport;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime
+{
+    /// <summary>
+    ///     Represents a discovered plugin that is available for installation.
+    /// </summary>
+    public sealed class AvailablePlugin
+    {
+        private readonly IPluginDiscoverer pluginDiscoverer;
+        private readonly IPluginFetcher pluginFetcher;
+
+        /// <summary>
+        ///     Creates an instance of <see cref="AvailablePlugin"/>
+        /// </summary>
+        /// <param name="pluginInfo">
+        ///     The <see cref="AvailablePluginInfo"/> object cotaining information about this plugin.
+        /// </param>
+        /// <param name="discoverer">
+        ///     The <see cref="IPluginDiscoverer"/> this plugin is discovered by.
+        /// </param>
+        /// <param name="fetcher">
+        ///     The <see cref="IPluginFetcher"/> this plugin uses to fetch plugin content.
+        /// </param>
+        internal AvailablePlugin(
+            AvailablePluginInfo pluginInfo,
+            IPluginDiscoverer discoverer,
+            IPluginFetcher fetcher)
+        {
+            this.AvailablePluginInfo = pluginInfo;
+            this.pluginDiscoverer = discoverer;
+            this.pluginFetcher = fetcher;
+        }
+
+        /// <summary>
+        ///     Gets the <see cref="AvailablePluginInfo"/> associated with this plugin.
+        /// </summary>
+        public AvailablePluginInfo AvailablePluginInfo { get; }
```

Comment:
nit: I would personally name this property just "Info." Accessing this as `AvailablePlugin.AvailablePluginInfo` is a bit redundant IMO